src/main.rs

```diff
@@ -402,89 +457,140 @@ fn get_args(matches: &ArgMatches) -> CliArgs {
         }
         None => VolumeType::Local,
     };
-    CliArgs {
-        end_point,
+    volume_type
+}
+
+/// Generate the metadata
+#[must_use]
+pub fn gen_metadata<'a>(name: &str) -> App<'a, 'a> {
+    let arg_map = get_default_arg_map();
+    SubCommand::with_name(name)
+        .arg(get_map(&arg_map, MOUNT_POINT_ARG_NAME))
+        .arg(get_map(&arg_map, WORKER_PORT_ARG_NAME))
+        .arg(get_map(&arg_map, NODE_ID_ARG_NAME))
+        .arg(get_map(&arg_map, NODE_IP_ARG_NAME))
+        .arg(get_map(&arg_map, ETCD_ADDRESS_ARG_NAME))
+}
+
+/// Parse the metadata
+pub fn parse_metadata(
+    matches: &ArgMatches,
+    role_name: RunAsRole,
+) -> Result<MetaData, common::error::DatenLordError> {
+    let etcd_delegate = EtcdDelegate::new(get_etcd_address_vec(matches))?;
+    let worker_port = get_worker_port(matches);
+    let node_id = get_node_id(matches);
+    let ip_address = get_ip_address(matches, &node_id);
+    let mount_dir = get_mount_dir(matches);
+
+    csi::build_meta_data(
         worker_port,
         node_id,
         ip_address,
-        driver_name,
-        mount_dir: mount_dir.to_owned(),
-        run_as,
-        etcd_address_vec,
-        scheduler_extender_port,
-        cache_capacity,
-        server_port: server_port.to_owned(),
-        volume_info: volume_info.to_owned(),
-        volume_type,
+        mount_dir.to_owned(),
+        role_name,
+        etcd_delegate,
+    )
+}
+
+/// Get command from default map
+fn get_map<'a>(map: &HashMap<&str, Arg<'a, 'a>>, name: &str) -> Arg<'a, 'a> {
+    if let Some(res) = map.get(name) {
+        res.clone()
+    } else {
+        panic!("Get default command fail");
```

Comment:
provide more information like panic!("Failed to get default argument {}", name);

Reply:
Done